homework-3/app.js

```diff
@@ -0,0 +1,106 @@
+const express = require("express");
+const bodyParser = require("body-parser");
+const uuid = require("uuid");
+
+const app = express();
+
+const fs = require("fs");
+const { json } = require("express");
+
+const port = 5000;
+
+const db = "./homework-3/db.json";
+
+app.use(bodyParser.urlencoded({ extended: false }));
+
+app.use(bodyParser.json());
+
+app.get("/products", (req, res) => {
+  fs.readFile(db, "utf8", (err, data) => {
+    if (err) throw err;
+    else {
+      let products = JSON.parse(data);
+      res.send(products);
+    }
+  });
+});
+
+app.post("/products", (req, res) => {
+  const { product } = req.body;
+  const parseProduct = JSON.stringify(product);
+  let productsPost;
+  fs.readFile(db, json, (err, data) => {
+    if (err) throw err;
+    else {
+      let products = JSON.parse(data);
+      products.push({ ...product, id: uuid.v4() });
+      productsPost = JSON.stringify(products);
+    }
+    fs.writeFile(db, productsPost, (err) => {
```

Comment:
you don't need to write all products to variable `productsPost`, just do `fs.writeFile(db, JSON.stringify(products),...`